sql/core/src/test/resources/sql-tests/inputs/pgSQL/strings.sql

```diff
@@ -0,0 +1,656 @@
+
+-- SELECT OVERLAY('yabadoo' PLACING 'daba' FROM 5 FOR 0) AS "yabadabadoo";
+
+-- SELECT OVERLAY('babosa' PLACING 'ubb' FROM 2 FOR 4) AS "bubba";
+
+--
+-- test LIKE
+-- Be sure to form every test as a LIKE/NOT LIKE pair.
+--
+
+-- simplest examples
+-- E061-04 like predicate
+SELECT 'hawkeye' LIKE 'h%' AS `true`;
+SELECT 'hawkeye' NOT LIKE 'h%' AS `false`;
+
+SELECT 'hawkeye' LIKE 'H%' AS `false`;
+SELECT 'hawkeye' NOT LIKE 'H%' AS `true`;
+
+SELECT 'hawkeye' LIKE 'indio%' AS `false`;
+SELECT 'hawkeye' NOT LIKE 'indio%' AS `true`;
+
+SELECT 'hawkeye' LIKE 'h%eye' AS `true`;
+SELECT 'hawkeye' NOT LIKE 'h%eye' AS `false`;
+
+SELECT 'indio' LIKE '_ndio' AS `true`;
+SELECT 'indio' NOT LIKE '_ndio' AS `false`;
+
+SELECT 'indio' LIKE 'in__o' AS `true`;
+SELECT 'indio' NOT LIKE 'in__o' AS `false`;
+
+SELECT 'indio' LIKE 'in_o' AS `false`;
+SELECT 'indio' NOT LIKE 'in_o' AS `true`;
+
+-- [SPARK-28083] ANSI SQL: LIKE predicate: ESCAPE clause
+-- unused escape character
+-- SELECT 'hawkeye' LIKE 'h%' ESCAPE '#' AS "true";
+-- SELECT 'hawkeye' NOT LIKE 'h%' ESCAPE '#' AS "false";
+
+-- SELECT 'indio' LIKE 'ind_o' ESCAPE '$' AS "true";
+-- SELECT 'indio' NOT LIKE 'ind_o' ESCAPE '$' AS "false";
+
+-- escape character
+-- E061-05 like predicate with escape clause
+-- SELECT 'h%' LIKE 'h#%' ESCAPE '#' AS "true";
+-- SELECT 'h%' NOT LIKE 'h#%' ESCAPE '#' AS "false";
+
+-- SELECT 'h%wkeye' LIKE 'h#%' ESCAPE '#' AS "false";
+-- SELECT 'h%wkeye' NOT LIKE 'h#%' ESCAPE '#' AS "true";
+
+-- SELECT 'h%wkeye' LIKE 'h#%%' ESCAPE '#' AS "true";
+-- SELECT 'h%wkeye' NOT LIKE 'h#%%' ESCAPE '#' AS "false";
+
+-- SELECT 'h%awkeye' LIKE 'h#%a%k%e' ESCAPE '#' AS "true";
+-- SELECT 'h%awkeye' NOT LIKE 'h#%a%k%e' ESCAPE '#' AS "false";
+
+-- SELECT 'indio' LIKE '_ndio' ESCAPE '$' AS "true";
+-- SELECT 'indio' NOT LIKE '_ndio' ESCAPE '$' AS "false";
+
+-- SELECT 'i_dio' LIKE 'i$_d_o' ESCAPE '$' AS "true";
+-- SELECT 'i_dio' NOT LIKE 'i$_d_o' ESCAPE '$' AS "false";
+
+-- SELECT 'i_dio' LIKE 'i$_nd_o' ESCAPE '$' AS "false";
+-- SELECT 'i_dio' NOT LIKE 'i$_nd_o' ESCAPE '$' AS "true";
+
+-- SELECT 'i_dio' LIKE 'i$_d%o' ESCAPE '$' AS "true";
+-- SELECT 'i_dio' NOT LIKE 'i$_d%o' ESCAPE '$' AS "false";
+
+-- escape character same as pattern character
+-- SELECT 'maca' LIKE 'm%aca' ESCAPE '%' AS "true";
+-- SELECT 'maca' NOT LIKE 'm%aca' ESCAPE '%' AS "false";
+
+-- SELECT 'ma%a' LIKE 'm%a%%a' ESCAPE '%' AS "true";
+-- SELECT 'ma%a' NOT LIKE 'm%a%%a' ESCAPE '%' AS "false";
+
+-- SELECT 'bear' LIKE 'b_ear' ESCAPE '_' AS "true";
+-- SELECT 'bear' NOT LIKE 'b_ear' ESCAPE '_' AS "false";
+
+-- SELECT 'be_r' LIKE 'b_e__r' ESCAPE '_' AS "true";
+-- SELECT 'be_r' NOT LIKE 'b_e__r' ESCAPE '_' AS "false";
+
+-- SELECT 'be_r' LIKE '__e__r' ESCAPE '_' AS "false";
+-- SELECT 'be_r' NOT LIKE '__e__r' ESCAPE '_' AS "true";
+
+--
+-- test ILIKE (case-insensitive LIKE)
+-- Be sure to form every test as an ILIKE/NOT ILIKE pair.
+--
+
+SELECT 'hawkeye' ILIKE 'h%' AS "true";
```

Comment:
Do we need to add `ILIKE` to `SPARK-27764`:
```
The key word ILIKE can be used instead of LIKE to make the match case-insensitive according to the active locale. This is not in the SQL standard but is a PostgreSQL extension.
```

https://www.postgresql.org/docs/11/functions-matching.html

Reply:
IMO we don't need to do cuz that's pg-specific.